pkg/common/mesh_config.go

```diff
@@ -0,0 +1,159 @@
+package common
```

Comment:
Why moving these functions to pkg/common?

Reply:
I reckoned they could be encapsulated and unit tested since are not dependent on client. pkg/common might not be the best destination, thought of _pkg/istio_ but the mesh config also belongs to OSSM... what do you have in mind?

Reply:
I think those functions are better kept flagged as private and the least opened to be reused at other places as possible. We don't want other things messing with the mesh config when this is responsibility of the Kuadrant controller.

Perhaps others who might be thinking on different (future) ways to manage instance of Kuadrant (e.g. CLI tool, add-on) may disagree. But at this stage I see the change as more than how we organise the code.

As for the unit tests, you can still test those functions in the controllers package even if they are not exported.

Reply:
I see your point and it makes sense. However, these functions are stateless and without the proper context of the Kuadrant controller reconciler, little they can mess with the actual mesh config. Also, even tho there's no planning on having a CLI or add-on,  we also have some functions that are strictly a responsibility of the kuadrant_controller, like manipulating the gateways. That being said, I'm fine with moving them back to the controller (or having them in any other package, maybe _mesh_config_?), since I might minimizing the scope of this change :)